### PR TITLE
[Bug] Fix the bug of grouping set core in both vec and no vec query engine

### LINKED ISSUE #7799: [Bug] Grouping Set CoreDump in TPCDS 77
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Version

master/vectorized

### What's Wrong?

core dump 

### What You Expected?

get result normally

### How to Reproduce?

_No response_

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by HappenLee**
# Proposed changes

Issue Number: close #7799 

## Problem Summary:

Describe the overview of changes.

## Checklist(Required)

1. Does it affect the original behavior: (Yes)
2. Has unit tests been added: (No Need)
3. Has document been added or modified: (No Need)
4. Does it need to update dependencies: (No)
5. Are there any changes that cannot be rolled back: (Yes)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/planner/RepeatNode.java [hunk 1]
```diff
@@ -131,7 +131,8 @@ public void init(Analyzer analyzer) throws UserException {
         // build tupleDesc according to child's tupleDesc info
         outputTupleDesc = groupingInfo.getVirtualTuple();
         //set aggregate nullable
-        for (Expr slot : groupByClause.getGroupingExprs()) {
+        for (int i = 0; i < groupByClause.getGroupingExprs().size() - 1; i++) {
```
Comment (EmmyMiao87) [inline fe/fe-core/src/main/java/org/apache/doris/planner/RepeatNode.java:134]: It doesn't look like it needs to be changed?

File: fe/fe-core/src/main/java/org/apache/doris/analysis/Analyzer.java [hunk 2]
```diff
@@ -722,7 +722,9 @@ public SlotDescriptor registerVirtualColumnRef(String colName, Type type, TupleD
         result = addSlotDescriptor(tupleDescriptor);
         Column col = new Column(colName, type);
         result.setColumn(col);
-        result.setIsNullable(true);
+        // now virtual slot: only use in grouping set to generate grouping id,
+        // so it should always is not nullable
+        result.setIsNullable(false);
```
Comment (EmmyMiao87) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/Analyzer.java:727]: It is better to use the `isNullable` of the column

File: fe/fe-core/src/main/java/org/apache/doris/planner/RepeatNode.java (hunk 1 above)
Comment (HappenLee) [reply fe/fe-core/src/main/java/org/apache/doris/planner/RepeatNode.java:134]: The last grouping expr is virtual grouping id， not need to set nullable

**Review by EmmyMiao87 — DISMISSED** [review]
LGTM

**Comment by morningman** [comment]
@HappenLee Please resolve the conflict

**Review by morningman — DISMISSED** [review]

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**